src/backend/joanie/core/serializers/admin.py

```diff
@@ -874,6 +884,14 @@ def get_recipient_address(self, invoice):
         """Return the serialized recipient address."""
         return f"{invoice.recipient_address.full_name}\n{invoice.recipient_address.full_address}"
 
+    def get_children(self, invoice):
+        """Return the serialized children invoices."""
+        return AdminInvoiceSerializer(invoice.children.all(), many=True).data
```

Comment:
If no recursion is needed, (a child could not have children), we can remove this method and fix the schema by using a base serializer:
``` python
class BaseAdminInvoiceSerializer(serializers.ModelSerializer):
    """Read only base serializer for Invoice model."""

    recipient_address = serializers.SerializerMethodField(read_only=True)
    total = serializers.DecimalField(
        coerce_to_string=False, decimal_places=2, max_digits=9
    )
    total_currency = serializers.SerializerMethodField(read_only=True)

    class Meta:
        model = payment_models.Invoice
        fields = [
            "balance",
            "created_on",
            "invoiced_balance",
            "recipient_address",
            "reference",
            "state",
            "transactions_balance",
            "total",
            "total_currency",
            "type",
            "updated_on",
        ]
        read_only_fields = fields

    def get_recipient_address(self, invoice):
        """Return the serialized recipient address."""
        return f"{invoice.recipient_address.full_name}\n{invoice.recipient_address.full_address}"

    def get_total_currency(self, *args, **kwargs) -> str:
        """Return the code of currency used by the instance"""
        return settings.DEFAULT_CURRENCY


class AdminInvoiceSerializer(BaseAdminInvoiceSerializer):
    """Read only serializer for Invoice model."""

    children = BaseAdminInvoiceSerializer(read_only=True, many=True)

    class Meta:
        model = BaseAdminInvoiceSerializer.Meta.model
        fields = [*BaseAdminInvoiceSerializer.Meta.fields, "children"]
        read_only_fields = fields
```

Reply:
`@extend_schema_serializer(component_name="AdminInvoiceSerializer")` has been added to remove `BaseAdminInvoice` from the generated schema.

Reply:
Yup indeed, for now, only a `main_invoice` can have children.

Reply:
(Is something like that)(86ecfa8d93c1e2c4076f7f8ec6d87b07e83702c1) looks good ?

Reply:
Great :+1: